hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ColumnStatsIndexSupport.scala

```diff
@@ -332,7 +331,7 @@ class ColumnStatsIndexSupport(spark: SparkSession,
     colStatsDF.select(targetColumnStatsIndexColumns.map(col): _*)
   }
 
-  def loadColumnStatsIndexRecords(targetColumns: Seq[String], shouldReadInMemory: Boolean): HoodieData[HoodieMetadataColumnStats] = {
+  def loadColumnStatsIndexRecords(targetColumns: Seq[String], prunedPartitions: Option[Set[String]] = None, shouldReadInMemory: Boolean): HoodieData[HoodieMetadataColumnStats] = {
```

Comment:
let's add UT for this method or the caller of this method? Just want to ensure that partitions fetched from `getPrunedPartitionsAndFileNames` and encodng done here are correct. In the current test in ths patch, I see `prunedPartitions` is set as None here https://github.com/apache/hudi/pull/12124/files#diff-d639b070298814506c692353524fecf43ca61a533f784f1b989dbc45ce67bbd5R197